src/chocolatey/infrastructure.app/nuget/NugetList.cs

```diff
@@ -465,6 +460,72 @@ public static IPackageSearchMetadata FindPackage(
             return packagesList.OrderByDescending(p => p.Identity.Version).FirstOrDefault();
         }
 
+        private static IOrderedEnumerable<IPackageSearchMetadata> ApplyPackageSort(IEnumerable<IPackageSearchMetadata> query, domain.PackageOrder orderBy)
+        {
+            switch (orderBy)
+            {
+                case domain.PackageOrder.LastPublished:
+                    return query
+                        .OrderByDescending(q => q.Published)
+                        .ThenBy(q => q.Identity.Id)
+                        .ThenBy(q => q.Identity.Version);
+
+                case domain.PackageOrder.Id:
+                    return query.OrderBy(q => q.Identity.Id)
+                        .ThenBy(q => q.Title)
+                        .ThenBy(q => q.Identity.Version);
+
+                case domain.PackageOrder.Popularity:
+                    return query
+                        .OrderByDescending(q => q.DownloadCount)
+                        .ThenByDescending(q => q.VersionDownloadCount)
+                        .ThenBy(q => q.Identity.Id);
+
+                case domain.PackageOrder.Title:
+                    return query
+                        .OrderBy(q => q.Title)
+                        .ThenBy(q => q.Identity.Id)
+                        .ThenBy(q => q.Identity.Version);
+
+                default:

```

Comment:
This "should" never happen, right? Since we are defaulting to `PackageOrder.Id` in `ChocolateyConfiguration` constructor.

Reply:
It "should" and will, since the PackageOrder can be "Unsorted". In this case it will go into this default statement. It is also used for a catch all for the future as a safe-guard as well.